src/field/field2d.cxx

```diff
@@ -174,12 +174,13 @@ const IndexRange Field2D::region(REGION rgn) const {
   };
 }
 
-BoutReal& Field2D::operator[](const Ind3D &d) {
-  return data[fieldmesh->map3Dto2D(d)];
+// Not in header because we need to access fieldmesh
+inline BoutReal& Field2D::operator[](const Ind3D &d) {
```

Comment:
Isn't `fieldmesh` part of the class? Should be find in the header I would think.

Reply:
`Mesh` is just forward declared in the Field2D header to avoid circular dependencies, but `mesh.hxx` is included (via `globals.hxx`) in the implementation file.

Reply:
Ah yes, tricky. Since it has to call a function `map3Dto2D` anyway perhaps it's not a problem that it won't be inlined.